client/state/lib/middleware.js

```diff
@@ -145,7 +145,7 @@ const handler = ( dispatch, action, getState ) => {
 
 		case SITE_DELETE_RECEIVE:
 		case JETPACK_DISCONNECT_RECEIVE:
-			user().fetch();
+			dispatch( fetchCurrentUser() );
```

Comment:
How hard would it be to move the refetch out of the middleware completely? And doing in instead at the places where the `*_RECEIVE` actions are dispatched?

Reply:
It's pretty straightforward. Seems like something for another PR though. I'm happy to follow up immediately after merging this one unless you feel strongly that it should happen in this PR.

Reply:
OK :+1:

Reply:
Moving the user fetching out of the state middleware: #53817